Slightly improved

angerman opened the pull request:
This makes the whole thing a bit more configurable.

We should probably read the config from a file? (e.g. a `settings` file, like the one ghc has?)

angerman committed:
Slightly improved

hvr commented on src/Main.hs:
```diff
@@ -13,19 +13,42 @@ import qualified Data.Set as Set
+                                      ".tmpl"
+                                      "cabal.cfg"
+                                      "packages"
+                                      "hackage.haskell.org"
```
Please make these configurable via CLI (e.g. optparse-applciative), as they're obviously specific to the installation (i.e. gtar/urls). If you want, you can make the foldernames configurable as well, but it's not that essential (as one can symlink as workaround).

angerman replied:
I think this is resolved now?

angerman committed:
Drop hey

angerman committed:
Adds optparse

angerman committed:
Drop requirement for `cabal.cfg`.

As we have the knowledge about the content of `cabal.cfg` anyhow. We can just
compute it in a temprary directory and use it, instead of requireing the
user to provide redundent information, which could lead to inconsistncies and
hard to find issues / bad experience.

Fixes #3.

angerman commented:
This now includes a fix for #3.

angerman commented:
I believe this is now ready to merge. Let me know what you think @hvr.

hvr commented on tool.cabal:
```diff
@@ -21,6 +21,7 @@ executable tool
                , text            ^>= 1.2.2.2
                , system-filepath ^>= 0.4.13.4
                , containers      ^>= 0.5.6.2
+               , optparse-applicative
```
...I think you forgot a version annotation here... :-)

angerman committed:
add optparse-applicative version

angerman committed:
meh. tool.cabal, not cabal.config

hvr commented:
thanks!
